jerry-core/ecma/base/ecma-globals.h

```diff
@@ -227,56 +306,61 @@ typedef struct
 } ecma_getter_setter_pointers_t;
 
 /**
- * Description of ecma-property
+ * Property data.
  */
-typedef struct ecma_property_t
+typedef union
 {
-  /** Compressed pointer to next property */
-  mem_cpointer_t next_property_p;
+  ecma_value_t value; /**< value of a property */
+  ecma_getter_setter_pointers_t getter_setter_pair; /**< getter setter pair */
+} ecma_property_value_t;
 
-  /** Property's flags (ecma_property_flags_t) */
-  uint8_t flags;
+/**
+ * Property pair.
+ */
+typedef struct
+{
+  ecma_property_header_t header; /**< header of the property */
+  ecma_property_value_t first_value; /**< first property value */
+  ecma_property_value_t second_value; /**< second property value */
+  mem_cpointer_t first_name_cp; /**< first property name */
+  mem_cpointer_t second_name_cp; /**< second property name */
```

Comment:
But if the structure remains the way proposed in the PR then it will be hardly extensible (effectively meaning no chance for "try out other layouts later"). What's blocking moving to a 

```c
typedef struct
{
  ecma_property_t types[2]; /**< property type bytes */
} ecma_property_header_t;

typedef struct
{
  ecma_property_header_t header; /**< header of the property */
  ecma_property_value_t values[2]; /**< property values */
  mem_cpointer_t name_cp[2]; /**< property names */
} ecma_property_pair_t;
```

style? This has really no semantic difference to the PR's approach.